src/app/components/challenge/challengesubmit/challengesubmit.component.ts

```diff
@@ -484,6 +498,47 @@ export class ChallengesubmitComponent implements OnInit {
   }
 
   validateInput(inputValue) {
-    this.inputFile = inputValue === null;
+    const regex = new RegExp(/https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)?/);
+    const validExtensions = ['json', 'zip', 'csv'];
+    if (this.fileboxchecked === true) {
+      this.inputFile = inputValue === null;
+    } else if (this.urlboxchecked = true) {
+      const extension = inputValue.split('.').pop();
+      this.inputErrorMessage = '';
+      if (regex.test(inputValue) && validExtensions.includes(extension)) {
+        this.inputErrorMessage = '';
+        this.inputFile = false;
+        this.validFileUrl = true;
+
+      } else {
+        this.inputErrorMessage = 'Please enter a valid URL which ends in json, zip or csv file extension!';
+        this.inputFile = true;
+        this.validFileUrl = false;
+      }
+    }
+  }
+
+  onSubmissionOptionChange(value: any) {
+    const phasebutton = <HTMLScriptElement>document.getElementById('Submission_option_hidden');
+    const formhidden = <HTMLScriptElement>document.getElementById('form_hidden');
+    const uploadbutton = <HTMLScriptElement>document.getElementById('uploadbutton');
+    const url_field = <HTMLScriptElement>document.getElementById('url_field');
```

Comment:
Instead of all these can we use variables and toggle them inside `*ngIf` in html based on the selection of submission option?